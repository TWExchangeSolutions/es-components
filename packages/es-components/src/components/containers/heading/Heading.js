import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

// should eventually come from theme
const fontSize = {
  1: '44.78976px',
  2: '37.3248px',
  3: '31.104px',
  4: '25.92px',
  5: '21.6px',
  6: '18px'
};

const BaseHeading = css`
  color: inherit;
  font-size: ${props => fontSize[props.adjustedSize]};
  font-weight: 300;
  line-height: 1.1;
  margin-bottom: 12.5px;
  margin-top: 25px;

  small {
    font-size: ${props => (props.adjustedSize > 3 ? '75%' : '65%')};
    line-height: 1;
  }
`;

// most of these, along with the switch statement, can be removed when we
// upgrade to styled-components v4 and can use the "as" prop
const Heading1 = styled.h1`
  ${BaseHeading};
`;

const Heading2 = styled.h2`
  ${BaseHeading};
`;

const Heading3 = styled.h3`
  ${BaseHeading};
`;

const Heading4 = styled.h4`
  ${BaseHeading};
`;

const Heading5 = styled.h5`
  ${BaseHeading};
`;

const Heading6 = styled.h6`
  ${BaseHeading};
`;

function Heading({ children, level, size, ...other }) {
  const adjustedSize = size || level;

  let HeaderStyled;
  switch (level) {
    case 2:
      HeaderStyled = Heading2;
      break;
    case 3:
      HeaderStyled = Heading3;
      break;
    case 4:
      HeaderStyled = Heading4;
      break;
    case 5:
      HeaderStyled = Heading5;
      break;
    case 6:
      HeaderStyled = Heading6;
      break;
    default:
      HeaderStyled = Heading1;
      break;
  }
  return (
    <HeaderStyled adjustedSize={adjustedSize} {...other}>
      {children}
    </HeaderStyled>
  );
}

Heading.propTypes = {
  children: PropTypes.node,
  /** Heading level element */
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
  /** Override the default font size with another level */
  size: PropTypes.oneOf([1, 2, 3, 4, 5, 6])
};

Heading.defaultProps = {
  children: undefined,
  size: undefined
};

export default Heading;
