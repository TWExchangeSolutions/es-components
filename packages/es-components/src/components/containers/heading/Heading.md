A `Heading` component can be used for various heading levels. Wrap secondary text in `<small>` tags.

```
<div>
  <Heading level={1}>h1. Heading <small>Secondary Text</small></Heading>
  <Heading level={2}>h2. Heading <small>Secondary Text</small></Heading>
  <Heading level={3}>h3. Heading <small>Secondary Text</small></Heading>
  <Heading level={4}>h4. Heading <small>Secondary Text</small></Heading>
  <Heading level={5}>h5. Heading <small>Secondary Text</small></Heading>
  <Heading level={6}>h6. Heading <small>Secondary Text</small></Heading>
</div>
```

The font size may be overridden by that of another level by providing the `size` prop.

```
<div>
  <Heading level={1} size={3}>h1. Heading in h3 size</Heading>
</div>
```
