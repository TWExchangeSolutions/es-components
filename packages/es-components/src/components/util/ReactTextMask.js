import React from 'react';
import ReactTextMask from 'react-text-mask';

// the component was using onInput which made testing more difficult.
// switched to onChange and put in a pull request so hopefully we can
// get rid of this soon.
ReactTextMask.prototype.render = function render() {
  const props = { ...this.props };

  delete props.mask;
  delete props.guide;
  delete props.pipe;
  delete props.placeholderChar;
  delete props.keepCharPositions;
  delete props.onChange;
  delete props.value;
  delete props.showMask;

  /* eslint-disable jsx-a11y/use-onblur-not-onchange */
  return (
    <input
      {...props}
      onChange={event => this.onChange(event)}
      defaultValue={this.props.value}
      ref={inputElement => (this.inputElement = inputElement)}
    />
    /* eslint-enable */
  );
};

export default ReactTextMask;
