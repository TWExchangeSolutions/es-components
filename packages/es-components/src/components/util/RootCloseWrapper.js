import React, { useRef } from 'react';
import { useRootClose } from 'react-overlays'

// eslint-disable-next-line react/prop-types
const RootCloseWrapper = ({children, onRootClose, disabled}) => {
    const ref = useRef();
    useRootClose(ref, onRootClose, {
        disabled
    });

    return (
      <div ref={ref} className="root-close-wrapper">
        {children}
      </div>
    );
}

export default RootCloseWrapper;