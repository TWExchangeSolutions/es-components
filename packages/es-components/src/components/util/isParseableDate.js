import parse from 'date-fns/parse';
import isValid from 'date-fns/is_valid';

export default function isParseableDate(date) {
  const parsedDate = parse(date);
  return isValid(parsedDate);
}
