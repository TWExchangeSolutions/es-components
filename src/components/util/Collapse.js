import React from 'react';
import Collapse from 'react-smooth-collapse';

/*
  The author of react-smooth-collapse changed things and broke
  support for additional properties. I opened an issue in their
  repo, and for now I'm adding this override to keep the markup
  in Drawer as clean as I can by allowing the passing of id
  and className props.
*/
Collapse.prototype.render = function render() {
  const visibleWhenClosed = this._visibleWhenClosed();
  const { height, fullyClosed, hasBeenVisibleBefore } = this.state;
  const {
    heightTransition,
    children,
    className,
    id
  } = this.props;

  const innerEl = hasBeenVisibleBefore ?
    <div ref="inner" style={{ overflow: 'hidden' }}>
      { children }
    </div>
    : null;

  return (
    <div
      ref="main"
      id={id}
      className={className}
      style={{
        height, overflow: 'hidden',
        display: (fullyClosed && !visibleWhenClosed) ? 'none' : null,
        transition: `height ${heightTransition}`
      }}
    >
      {innerEl}
    </div>
  );
};

export default Collapse;
