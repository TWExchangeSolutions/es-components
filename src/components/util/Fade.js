import React from 'react';
import PropTypes from 'prop-types';
import Transition from 'react-transition-group/Transition';

const Fade = ({ children, duration, opacity, withWrapper, ...otherProps }) => {
  const transitionStyles = {
    transition: `opacity ${duration}ms ease-in-out`,
    opacity: 0
  };

  const stateStyles = {
    entering: { opacity },
    entered: { opacity }
  };

  // the withWrapper option is to support components using the Overlay/Position
  // react-overlay components, which don't properly pass children style props
  // as of the 0.8.3
  return withWrapper ? (
    <Transition {...otherProps} timeout={duration}>
      {(state, innerProps) => (
        <div style={{ ...transitionStyles, ...stateStyles[state] }}>
          {children}
        </div>
      )}
    </Transition>
  ) : (
    <Transition {...otherProps} timeout={duration}>
      {(state, innerProps) =>
        React.cloneElement(children, {
          ...innerProps,
          style: {
            ...children.props.style,
            ...transitionStyles,
            ...stateStyles[state]
          }
        })}
    </Transition>
  );
};

Fade.propTypes = {
  children: PropTypes.any.isRequired,
  duration: PropTypes.number,
  opacity: PropTypes.number,
  withWrapper: PropTypes.bool
};

Fade.defaultProps = {
  duration: 300,
  opacity: 1,
  withWrapper: false
};

export default Fade;
