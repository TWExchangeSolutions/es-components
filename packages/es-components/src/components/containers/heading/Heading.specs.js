/* eslint-env jest */

import React from 'react';
import { renderWithTheme } from '../../util/test-utils';

import Heading from './Heading';

it('renders proper heading level', () => {
  const { queryByText } = renderWithTheme(
    <Heading level={1}>Heading level 1</Heading>
  );

  expect(queryByText('Heading level 1')).not.toBeNull();
});

it('renders heading level with another size', () => {
  const { container } = renderWithTheme(
    <div>
      <Heading level={1}>Heading level 1</Heading>
      <Heading level={3}>Heading level 3</Heading>
      <Heading level={3} size={1}>
        Heading level 3 size 1
      </Heading>
    </div>
  );

  expect(container).toMatchSnapshot();
});
