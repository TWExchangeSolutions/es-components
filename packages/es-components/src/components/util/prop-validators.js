import isParseableDate from './isParseableDate';

export function isPropValidDate(props, propName, componentName) {
  const date = props[propName];
  if (date) {
    if (!isParseableDate(date)) {
      return new Error(`The pre-selected date is not valid. Please enter a parseable date.
        See http://en.wikipedia.org/wiki/ISO_8601 for valid date formats.
      `);
    }
    return null;
  }
  return null;
}
