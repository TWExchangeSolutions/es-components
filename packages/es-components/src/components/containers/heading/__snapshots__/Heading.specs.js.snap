// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heading level with another size 1`] = `
<div>
  <div>
    <h1
      class="sc-bdVaJa epHIGJ"
    >
      Heading level 1
    </h1>
    <h3
      class="sc-htpNat fscdUm"
    >
      Heading level 3
    </h3>
    <h3
      class="sc-htpNat eUqHgu"
    >
      Heading level 3 size 1
    </h3>
  </div>
</div>
`;
